pdb2pqr/biomolecule.py

```diff
@@ -748,7 +748,10 @@ def apply_name_scheme(self, forcefield_):
                         if (residue.is_n_term or residue.is_c_term) and (
                             rname != residue.name
                         ):
-                            rname = residue.name
+                            if len(rname) == 4 and rname[0] in ("C", "N"):
```

Comment:
I don't understand this.  Can @stefdoerr or someone add documentation to the code about why this change is needed?

Reply:
The change is needed because as you can see in the linked issue (and test file in this PR) pdb2pqr renames terminal protonated residues back to their base form (CYX->CYS, LYN->LYS etc) which breaks any software coming after pdb2pqr like tleap since the hydrogens are wrong for the given resname.

As to if this change is the best solution, beyond passing the tests, I don't know. I don't have the global view you have of the software so I patch the issues where I see them. I assume that for example maybe the `residue.name` property could be set to the specific protonation when it's applied but I won't dig that much in the internals.

Reply:
Also if you enable by default the resname comparison you will see that many of the other tests are broken. They have the wrong resnames for the specific protonations. I made the resname comparison optional to not bother updating all the broken tests, I leave that to you for another PR.

Reply:
Can you please add these comments to the code, rather than the PR?  Thanks.

Reply:
Done